From 755b1b93035cd18ee95521e4108304e71e0a3219 Mon Sep 17 00:00:00 2001
From: Fabian Brandt-Tumescheit
 <fabian.brandt-tumescheit@informatik.hu-berlin.de>
Date: Thu, 8 Jul 2021 12:39:11 +0200
Subject: [PATCH] Pass conda-python version

---
 setup.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 906ef4419..ea3df1664 100644
--- a/setup.py
+++ b/setup.py
@@ -38,6 +38,9 @@ os_soabi = sysconfig.get_config_var('SOABI')
 if os_soabi is None:
 	os_soabi =  sysconfig.get_config_var('EXT_SUFFIX').split(".")[1] # get_config_var('SOABI') is None on win32-systems
 
+if "PYTHON" in os.environ:
+    python_exe = os.environ["PYTHON"]
+
 import os
 import subprocess #calling cmake, make and cython
 
@@ -155,7 +158,9 @@ def buildNetworKit(install_prefix, externalCore=False, externalTlx=None, withTes
 	comp_cmd.append(os.getcwd()) #call CMakeLists.txt from networkit root
 	if rpath:
 		comp_cmd.append("-DNETWORKIT_PYTHON_RPATH="+rpath)
-	# Run cmake
+	if pythowqn_exe:
+		comp_cmd.append("-DPython3_EXECUTABLE="+python_exe)
+    # Run cmake
 	print("initializing NetworKit compilation with: '{0}'".format(" ".join(comp_cmd)), flush=True)
 	if not subprocess.call(comp_cmd, cwd=buildDirectory) == 0:
 		print("cmake returned an error, exiting setup.py")
-- 
2.30.0

